src/test/java/org/fogbowcloud/manager/requests/api/local/http/ComputeOrdersControllerTest.java

```diff
@@ -48,132 +48,572 @@
+        InstanceStatus instanceStatus3 = new InstanceStatus(FAKE_ID_3, InstanceState.READY);
+        
+        List<InstanceStatus> computeStatusList = Arrays.asList(new InstanceStatus[] {instanceStatus1, instanceStatus2, instanceStatus3});
+        Mockito.doReturn(computeStatusList).when(this.facade).getAllInstancesStatus(Mockito.anyString(), Mockito.any(InstanceType.class));
+        
+        String COMPUTE_STATUS_ENDPOINT = COMPUTE_ENDPOINT + "/" + ComputeOrdersController.STATUS_ENDPOINT;
+        RequestBuilder requestBuilder = createRequestBuilder(HttpMethod.GET, COMPUTE_STATUS_ENDPOINT, getHttpHeaders(), "");
+        
+        // exercise
+        MvcResult result = this.mockMvc.perform(requestBuilder).andReturn();
+        
+        // verify
+        int expectedStatus = HttpStatus.OK.value();
+        Assert.assertEquals(expectedStatus, result.getResponse().getStatus());
+        
+        TypeToken<List<InstanceStatus>> token = new TypeToken<List<InstanceStatus>>(){};
+        List<InstanceStatus> resultList = new Gson().fromJson(result.getResponse().getContentAsString(), token.getType());
+        Assert.assertTrue(resultList.size() == 3);
+        Assert.assertEquals(FAKE_ID_1, resultList.get(0).getInstanceId());
+        Assert.assertEquals(FAKE_ID_2, resultList.get(1).getInstanceId());
+        Assert.assertEquals(FAKE_ID_3, resultList.get(2).getInstanceId());
+        
+        Mockito.verify(this.facade, Mockito.times(1)).getAllInstancesStatus(Mockito.anyString(), Mockito.any(InstanceType.class));
+    }
+    
+    
+    // test case: Request a compute by its id with an unauthenticated user. Check the response of request
+    // and the call of facade for get the compute.
     @Test
-    public void getNotFoundComputeById() throws Exception {
-        final String fakeId = "fake-Id-1";
-        String computeIdEndpoint = COMPUTE_ENDPOINT + "/" + fakeId;
-        doThrow(new InstanceNotFoundException()).when(this.facade).getCompute(anyString(), anyString());
+    public void testGetComputeByIdUnauthenticatedException() throws Exception {
+    	
+    	// set up
+    	final String FAKE_ID = "fake-Id-1";
+        String computeIdEndpoint = COMPUTE_ENDPOINT + "/" + FAKE_ID;
+        Mockito.doThrow(new UnauthenticatedUserException()).when(this.facade).getCompute(Mockito.anyString(), Mockito.anyString());
         RequestBuilder requestBuilder = createRequestBuilder(HttpMethod.GET, computeIdEndpoint, getHttpHeaders(), "");
-
+        
+        // exercise
         MvcResult result = this.mockMvc.perform(requestBuilder).andReturn();
-
+        
+        
+        // verify
+        int expectedStatus = HttpStatus.UNAUTHORIZED.value();
+        Assert.assertEquals(expectedStatus, result.getResponse().getStatus());
+        
+        Mockito.verify(this.facade, Mockito.times(1)).getCompute(Mockito.anyString(), Mockito.anyString());
+    }
+    
```

Comment:
extra blank line (ask @fgan1 or @arn3tt) for the code formatter they are using.

Reply:
I didn't get it. Blank line between assert and verify?

Reply:
306 307 maybe. check the whole code, there are more dead lines